extensions/ccui/base-classes/UIWidget.js

```diff
@@ -656,12 +665,27 @@ ccui.Widget = ccui.ProtectedNode.extend(/** @lends ccui.Widget# */{
      * @param {Boolean} enable  true if the widget is touch enabled, false if the widget is touch disabled.
      */
     setTouchEnabled: function (enable) {
-        if (this._touchEnabled === enable)
-            return;
+        if (enable == false) {
+            if (this._touchEnabled) {
+                cc.eventManager.removeListener(this._touchListener);
+            }
+        }
+
+        this._touchEnabled = enable;
 
-        this._touchEnabled = enable;                                  //TODO need consider remove and re-add.
-        if (this._touchEnabled) {
-            if(!this._touchListener)
+        // when the node is in the running scene
+        if (this._onRunningScene == true) {
+            _manageListener(enable);
+        }
+    },
+
+    /**
```

Comment:
This is a private function, you can use `//` comment instead of document comment